src/Unicorn/UI/Pipelines/GetContentEditorWarnings/SerializedWarning.cs

```diff
@@ -32,18 +33,35 @@ public virtual void Process(GetContentEditorWarningsArgs args)
 
 			var existingSitecoreItem = new ItemData(item);
 
-			var configuration = _configurations.FirstOrDefault(config => config.Resolve<IPredicate>().Includes(existingSitecoreItem).IsIncluded);
-			if (configuration != null)
+			IConfiguration foundConfiguration = null;
+			PredicateResult foundPredicateResult = null;
+
+			foreach (var configuration in _configurations)
+			{
+				var predicate = configuration.Resolve<IPredicate>();
+				var predicateResult = predicate.Includes(existingSitecoreItem);
+				if (predicateResult.IsIncluded)
+				{
+					foundConfiguration = configuration;
+					foundPredicateResult = predicateResult;
+					break;
+				}
+			}
+
+			if (foundConfiguration != null)
 			{
-				var evaluator = configuration.Resolve<IEvaluator>();
+				var evaluator = foundConfiguration.Resolve<IEvaluator>();
 
 				var warningObject = evaluator.EvaluateEditorWarning(item);
 
 				if (warningObject != null)
 				{
 					GetContentEditorWarningsArgs.ContentEditorWarning warning = args.Add();
 					warning.Title = warningObject.Title;
-					warning.Text = warningObject.Message;
+					if (Settings.GetBoolSetting("Unicorn.DevMode", true))
```

Comment:
This really should belong in the Evaluator, so that it retains full control over the message that is shown. It also would enable removing a lot of the added complexity from here as the predicate already has an `IConfiguration` instance :)

e.g. https://github.com/kamsar/Unicorn/blob/master/src/Unicorn/Evaluators/SerializedAsMasterEvaluator.cs#L84